crates/libtiny_tui/src/config.rs

```diff
@@ -28,6 +35,161 @@ pub(crate) struct Config {
     pub(crate) key_map: Option<KeyMap>,
 }
 
+impl Config {
+    /// Gets tab configs for `server`
+    /// Prioritizing configs under the server or using defaults
+    pub(crate) fn server_tab_configs(&self, server: &str) -> TabConfig {
+        let server_config = self
+            .servers
+            .iter()
+            .find(|s| s.addr == server)
+            .map(|s| &s.configs);
+        self.defaults.tab_configs.merge(server_config)
+    }
+
+    /// Gets tab configs for `chan` in `server`
+    /// Prioritizing configs under the chan, then the server, then the defaults
+    pub(crate) fn chan_tab_configs(&self, server: &str, chan: &ChanNameRef) -> TabConfig {
+        let tab_config = self
+            .servers
+            .iter()
+            .find(|s| s.addr == server)
+            .and_then(|s| s.join.iter().find(|c| chan == &c.name).map(|c| c.config));
+        self.server_tab_configs(server).merge(tab_config.as_ref())
+    }
+
+    pub(crate) fn user_tab_configs(&self) -> TabConfig {
+        TabConfig {
+            ignore: Some(false),
+            notifier: Some(Notifier::Messages),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+pub(crate) struct Server {
+    pub(crate) addr: String,
+    pub(crate) join: Vec<Chan>,
+    #[serde(flatten)]
+    pub(crate) configs: TabConfig,
+}
+
+#[derive(Debug, Default, Deserialize, PartialEq)]
+pub(crate) struct Defaults {
+    #[serde(flatten)]
+    pub(crate) tab_configs: TabConfig,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Chan {
+    pub name: ChanName,
+    pub config: TabConfig,
+}
+
+impl<'de> Deserialize<'de> for Chan {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct ChanVisitor;
+
+        impl<'de> Visitor<'de> for ChanVisitor {
+            type Value = Chan;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                write!(formatter, "a channel name with arguments")
+            }
+
+            fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                Chan::from_str(v).map_err(de::Error::custom)
+            }
+        }
+        deserializer.deserialize_str(ChanVisitor)
+    }
+}
+
+impl FromStr for Chan {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // Make sure channel starts with '#'
+        let s = if !s.starts_with('#') {
+            format!("#{}", s)
+        } else {
+            s.to_string()
+        };
```

Comment:
Not a big deal probably, but I wonder if we should expect user to get this right instead of inserting a `#` for them. The problem is channel names can contain multiple `#`s, not just one, but we assume one here.

Reply:
This is something I noticed other clients do, so I added it if there is no `#`. It seems like a nicer UX, but we don't need to have it.